src/integrationTest/java/com/mongodb/hibernate/BasicCrudIntegrationTests.java

```diff
@@ -169,6 +170,43 @@ void testSimpleDeletion() {
         }
     }
 
+    @Nested
+    class UpdateTests {
+
+        @Test
+        void testSimpleUpdate() {
+            var statementInspector = sessionFactoryScope.getStatementInspector(SQLStatementInspector.class);
+            statementInspector.clear();
+            sessionFactoryScope.inTransaction(session -> {
+                var book = new Book();
+                book.id = 1;
+                book.title = "War and Peace";
+                book.author = "Leo Tolstoy";
+                book.publishYear = 1867;
+                session.persist(book);
+                session.flush();
+
+                book.title = "Resurrection";
+                book.publishYear = 1899;
+                session.merge(book);
+            });
+
+            assertCollectionContainsOnly(
+                    BsonDocument.parse(
+                            """
+                            {"_id": 1, "author": "Leo Tolstoy", "publishYear": 1899, "title": "Resurrection"}\
+                            """));
+            assertThat(statementInspector.getSqlQueries())
+                    .containsExactly(
```

Comment:
Why do we use `SQLStatementInspector` here instead of checking the data in the DB? We don't use this approach for other tests in `BasicCrudIntegrationTests`, because we don't really care about the MQL generated, we only care about the data in the DB.

Reply:
The reason I used it is to verify that the end result is based on `insertion->update` steps, not one single insertion combining both. If this is a invalid concern, I could remove its usage.

Reply:
I think you are right. No need to test the implementation steps. Only the end result matters. The usage of `SQLStatementInspector` is deleted.

Reply:
> I used it is to verify that the end result is based on insertion->update steps, not one single insertion combining both

We know that it can't be just a single insertion combining both changes because we explicitly call `flush` before doing more changes.